src/bulkExport/bulkExportResults.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import AWS from 'aws-sdk';
+
+const EXPIRATION_TIME_SECONDS = 1800;
+const EXPORT_RESULTS_BUCKET = process.env.EXPORT_RESULTS_BUCKET || ' ';
+const EXPORT_RESULTS_SIGNER_ROLE_ARN = process.env.EXPORT_RESULTS_SIGNER_ROLE_ARN || '';
+
+const getFiles = async (jobId: string): Promise<string[]> => {
+    const s3 = new AWS.S3();
+
+    const listObjecsResult = await s3.listObjectsV2({ Bucket: EXPORT_RESULTS_BUCKET, Prefix: jobId }).promise();
+    return listObjecsResult.Contents!.map(x => x.Key!);
+};
+
+const signExportResults = async (keys: string[]): Promise<{ key: string; url: string }[]> => {
+    if (keys.length === 0) {
+        return [];
+    }
+    const sts = new AWS.STS();
+    const assumeRoleResponse = await sts
+        .assumeRole({
+            RoleArn: EXPORT_RESULTS_SIGNER_ROLE_ARN,
+            RoleSessionName: 'signBulkExportResults',
+            DurationSeconds: EXPIRATION_TIME_SECONDS,
+        })
+        .promise();
+
+    const s3 = new AWS.S3({
+        credentials: {
+            accessKeyId: assumeRoleResponse.Credentials!.AccessKeyId,
+            secretAccessKey: assumeRoleResponse.Credentials!.SecretAccessKey,
+            sessionToken: assumeRoleResponse.Credentials!.SessionToken,
+        },
+    });
+
+    return Promise.all(
+        keys.map(async key => ({
+            key,
+            url: await s3.getSignedUrlPromise('getObject', {
+                Bucket: EXPORT_RESULTS_BUCKET,
+                Key: key,
+                Expires: EXPIRATION_TIME_SECONDS,
+            }),
+        })),
+    );
+};
+
+const getResourceType = (key: string, jobId: string): string => {
+    const regex = new RegExp(`^${jobId}/([A-Za-z]+)-.*$`);
+    const match = regex.exec(key);
```

Comment:
When to use `regex.exec(key)` vs `key.match(regex)`? This is some somewhat clear guidance: https://stackoverflow.com/questions/9214754/what-is-the-difference-between-regexp-s-exec-function-and-string-s-match-fun/33139816#:~:text=exec()%20returns%20only%20the,See%20here%3A%20exec%2C%20match.&text=If%20your%20regex%20is%20global,then%20you%20must%20use%20exec.

But curious your thoughts

Reply:
afaik they are interchangeable unless you are doing global matching

> If the regular expression does not include the g flag, str.match() will return the same result as RegExp.exec().

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match